tests/testthat/test-validate_config.R

```diff
@@ -396,41 +396,22 @@ test_that("Duplicate property names are flagged during validation", {
   )
 })
 
-test_that("Version 5.1 validated accordingly", {
-  config_path <- testthat::test_path("testdata", "v5.1.0-tasks.json")
-  withr::with_options(
```

Comment:
We don't need the `with_option()` here because that's only used when creating config objects programmatically to enable setting the version once through an option instead of in every call to a `create_*()` fn.

In `validate_config()` we can use the `schema_version` argument directly to test a config file against diferent versions